docs(aio): added Loiane as GDE in contributors

loiane opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[X] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

gkalpak committed:
docs(upgrade): add API docs for `propagateDigest` (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
docs(upgrade): add API docs for `downgradeModule()` (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
docs: minor fixes in `docs-style-guide` (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
docs(upgrade): update API docs for upgrade/static (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
test: minor improvements in examples e2e tests script (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
docs(upgrade): add guide about `downgradeModule()` (#18487) (#18487)

PR Close #18487

PR Close #18487

petebacondarwin committed:
docs(upgrade): fix sub-ordered-list syntax (#18487) (#18487)

We must always use 1., 2. etc, to indicate ordered lists, even for sub-lists.
We can change the sublist to display as a., b. etc, via CSS.

PR Close #18487

PR Close #18487

petebacondarwin committed:
docs(upgrade): use a class for upgraded service (#18487) (#18487)

This makes the resulting use in Angular more ideomatic, since we can just
use the class type as the injection indicator.

PR Close #18487

PR Close #18487

rafaelss95 committed:
docs: fix typos in 'Httpclient' docs (#19127)

PR Close #19127

kapunahelewong committed:
docs(aio): tech edits to upgrade-lazy (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
build: upgrade jasmine (and related typings) to latest version (#19904)

With these changes, the types are a little stricter now and also not
compatible with Protractor's jasmine-like syntax. So, we have to also
use `@types/jasminewd2` for e2e tests (but not for non-e2e tests).

I also had to "augment" `@types/jasminewd2`, because the latest
typings from [DefinitelyTyped][1] do not reflect the fact that the
`jasminewd2` version (v2.1.0) currently used by Protractor supports
passing a `done` callback to a spec.

[1]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/566e0394859fdc1dc893658ccec6b06372d56a91/types/jasminewd2/index.d.ts#L9-L15

Fixes #23952
Closes #24733

PR Close #19904

MarkChrisLevy committed:
fix(compiler): support `.` in import statements. (#20634)

fix #20363

PR Close #20634

jamiekrug committed:
docs: fix typos referencing inline component styles (#22557)

PR Close #22557

sarunint committed:
docs(router): add `paramsInheritanceStrategy` documentation (#22590)

PR Close #22590

jamiekrug committed:
docs: fix typo in user input guide (#22630)

PR Close #22630

jamiekrug committed:
docs: fix incorrect forms selector references (#22631)

PR Close #22631

danielsogl committed:
docs(aio): unified string chaining (#22735)
PR Close #22735

danielsogl committed:
docs: unified string chaining (#22735)


PR Close #22735

danielsogl committed:
docs(aio): unified console.log single string style (#22737)
PR Close #22737

danielsogl committed:
docs: unified console.log single string style (#22737)


PR Close #22737

danielsogl committed:
docs: refactored ng-container code (#22742)


PR Close #22742

KevinVillela committed:
fix(core): use addCustomEqualityTester instead of overriding toEqual (#22983)

This propagates other custom equality testers added by users. Additionally, if
an Angular project is using jasmine 2.6+, it will allow Jasmine's custom object
differ to print out pretty test error messages.

fixes #22939

PR Close #22983

gkalpak committed:
build: fix windows scripts (#23121)

The `packages/core/src/animation/dsl.ts` symlink ws removed in #22692,
so `create-/remove-symlinks.sh` scripts for Windows should not try to
"fix" it.

PR Close #23121

mrmeku committed:
feat(bazel): Initial commit of protractor_web_test_suite (#24787)

Co-authored-by: Andrew Z Allen <me@andrewzallen.com>

PR Close #24787

Commit:
fix(router): add ability to recover from malformed url (#23283)

Fixes #21468

PR Close #23283

matsko committed:
test(animations): properly reference body node for SSR environments (#23300)

PR Close #23300

dpsthree committed:
docs: update Angular Boot Camp description (#23653)

PR Close #23653

realtomaszkula committed:
docs(aio): add cross field validation example (#23743)

PR Close #23743

Splaktar committed:
docs(aio): update contributors entry (#23786)

PR Close #23786

gregmagolan committed:
build(bazel): turn on preserve-symlinks (#24881)

This change turns on preserve-symlinks in nodejs to verify hermeticity of the Angular build.

BREAKING CHANGE: Use of @angular/bazel rules now requires calling ng_setup_workspace() in your WORKSPACE file.

For example:

local_repository(
    name = "angular",
    path = "node_modules/@angular/bazel",
)

load("@angular//:index.bzl", "ng_setup_workspace")

ng_setup_workspace()

PR Close #24881

Commit:
docs: add app.module to changed documents (#23876)

PR Close #23876

jbogarthyde committed:
docs: test doc for decorator templates (#23902) (#23902)

PR Close #23902

PR Close #23902

alcfeoh committed:
docs(aio): Add Angular Training to list of training companies (#23907)
PR Close #23907

arunkarthicksr committed:
docs: fix docregion in attribute directives for highlight directive (#23972)

Fixes #23503

PR Close #23972

kapunahelewong committed:
docs: clarify faqs about services (#24079)

PR Close #24079

petebacondarwin committed:
build(aio): compute breadcrums for all API doc types (#24000)

PR Close #24000

petebacondarwin committed:
build(aio): add github links to API doc members (#24000)

This change adds Github edit and view links to methods
and decorator options.

It is possible to add these to properties also but the
UI is rather tight as these are displayed in a table.

PR Close #24000

gkalpak committed:
feat(service-worker): add support for `?` in SW config globbing (#24105)

The globbing is used in the following sections:
- `assetGroups` > `resources` > `files`/`versionedFiles`
- `assetGroups` > `resources` > `urls`
- `dataGroups` > `urls`
- `navigationUrls`

Query params are ignored for `files`/`versionedFiles` and
`navigationUrls`, but they are still taken into account for
`assetGroups`/`dataGroups` `urls`. To avoid a breaking change, `?` is
matched literally for these patterns.

PR Close #24105

gkalpak committed:
refactor(service-worker): avoid unnecessary operations and remove unused code (#24127)

PR Close #24127

gkalpak committed:
fix(service-worker): avoid network requests when looking up hashed resources in cache (#24127)

PR Close #24127

jbogarthyde committed:
docs: test api doc for pipes (#24141)

PR Close #24141

rjamet committed:
docs: Add notes on manual sanitization to security guide (#24176)

Some users have remarked that we don't explain how to manually call
sanitization, so add a few lines on that.

PR Close #24176

alxhub committed:
feat(ivy): ngtsc compiles @Component, @Directive, @NgModule (#24427)

This change supports compilation of components, directives, and modules
within ngtsc. Support is not complete, but is enough to compile and test
//packages/core/test/bundling/todo in full AOT mode. Code size benefits
are not yet achieved as //packages/core itself does not get compiled, and
some decorators (e.g. @Input) are not stripped, leading to unwanted code
being retained by the tree-shaker. This will be improved in future commits.

PR Close #24427

brandonroberts committed:
docs(forms): update API reference for FormControl (#24255)

PR Close #24255

realtomaszkula committed:
docs(aio): address pr review issues (#23743)

PR Close #23743

gkalpak committed:
fix(language-service): do not overwrite native `Reflect` (#24299)

Fixes #21420

PR Close #24299

realtomaszkula committed:
docs(aio): fix issues suggested by Kara (#23743)

PR Close #23743

realtomaszkula committed:
docs(aio): fix issues suggested by Brandon (#23743)

PR Close #23743

brandonroberts committed:
docs(docs-infra): Update with review changes (#24255)

PR Close #24255

doggy8088 committed:
docs(aio): added a link to Angular Zero online course (Traditional Chinese) (#24228)

PR Close #24228

brandonroberts committed:
docs: More form control API references fixes (#24255)

PR Close #24255

brandonroberts committed:
docs: formatting (#24255)

PR Close #24255

brandonroberts committed:
docs: Added multicast to observable descriptions (#24255)

PR Close #24255

paulmojicatech committed:
docs: fix documention for attributes directive (#24367)

fix:update documentation for attributes directive to fix error

PR Close #24367

realtomaszkula committed:
docs(aio): fix adding to template driven forms (#23743)

PR Close #23743

loiane committed:
docs: added Loiane as GDE in contributors

mary-poppins commented:
You can preview 496cc09 at https://pr24423-496cc09.ngbuilds.io/.

brandonroberts committed:
docs: More form control API edits (#24255)

PR Close #24255

brandonroberts committed:
docs: More edits (#24255)

PR Close #24255

gkalpak committed:
test: remove unnecessary `yarn.lock` file (#24443)

PR Close #24443

gkalpak committed:
test(elements): test typings against TS 2.7 and 2.8 (#24443)

PR Close #24443

gkalpak committed:
docs(elements): mention `comp: elements` as a valid label (#24443)

PR Close #24443

fidian committed:
docs(forms): added missing backtick (#24451)

Fixed trivial markdown problem with a missing backtick.

PR Close #24451

Commit:
docs: fix typo in pipes guide (#24452)

PR Close #24452

alxhub committed:
feat(ivy): implement host bindings in JIT mode (#24479)

PR Close #24479

brandonroberts committed:
docs: Remove outdated Webpack guide and example (#24478)

A supporting Webpack guide will be introduced as part of the guidance
for ejecting from the Angular CLI.

Closes #23937

PR Close #24478

alxhub committed:
fix(ivy): keep JIT symbol table and r3_identifiers in sync (#24479)

At runtime in JIT mode, when the compiler writes a reference to a symbol that symbol
is resolved through a symbol table named angularCoreEnv in render3/jit/environment.
Previously, this symbol table was not kept up-to-date with the Ivy instruction set
and the names of symbols the compiler could reference.

This change brings the symbol table in sync, and also adds a test that verifies every
symbol the compiler can reference is available at runtime in the symbol table.

PR Close #24479

jbogarthyde committed:
docs: clarify wording in architecture overview (#24481)

Closes #23463
Closes #22158

PR Close #24481

alxhub committed:
style(ivy): defeat clang format issue (#24479)

clang-format (on mac) has taken a disliking to this particular line, and
rewrites one of the ɵ characters to an invalid Unicode sequence.

PR Close #24479

alxhub committed:
refactor(ivy): first pass at extracting ReflectionHost for abstract reflection (#24541)

ngtsc needs to reflect over code to property compile it. It performs operations
such as enumerating decorators on a type, reading metadata from constructor
parameters, etc.

Depending on the format (ES5, ES6, etc) of the underlying code, the AST
structures over which this reflection takes place can be very different. For
example, in TS/ES6 code `class` declarations are `ts.ClassDeclaration` nodes,
but in ES5 code they've been downleveled to `ts.VariableDeclaration` nodes that
are initialized to IIFEs that build up the classes being defined.

The ReflectionHost abstraction allows ngtsc to perform these operations without
directly querying the AST. Different implementations of ReflectionHost allow
support for different code formats.

PR Close #24541

cexbrayat committed:
test(common): rename keyvalue e2e test (#24489)

PR Close #24489

nicodh committed:
docs: fix unit tests in toh-pt6 (#24491)

Resolves #20373

PR Close #24491

FinnStutzenstein committed:
docs: consistent spacing in tutorial html files (#23105) (#24497)

- Removed surrounding spaces in interpolation expressions following the styleguide
- Consistant spacing of two spaces in html

Fixes #23105

PR Close #24497

pkozlowski-opensource committed:
fix(ivy): allow view and content queries to match the same element (#24507)

When creating content queries from a directive on an element we need to take into account
existing view queries. The same element can be reported to both content and view queries
so freshly created content queries must be combined with pre-existing view queries.

PR Close #24507

alexeagle committed:
build: update to Bazel 0.14.0 (#24512)

Includes a fix for out-of-memory condition which caused this to be
reverted yesterday.

PR Close #24512

mhevery committed:
fix(core): Injector correctly honors the @Self flag (#24520)

Injector was incorrectly returning instance from parent injector even
when `@Self` was specified.

PR Close #24520

gregmagolan committed:
test(bazel): fix flakey bazel integration e2e test (#24522)

PR Close #24522

petebacondarwin committed:
build(docs-infra): move overload short description above syntax (#24526)

PR Close #24526

riren committed:
docs(common): fix in the documentation of PUT (#24528)
PR Close #24528

petebacondarwin committed:
build(docs-infra): ensure all headings are sentence cased (#24527)

PR Close #24527

petebacondarwin committed:
build(docs-infra): support hiding constructors of injectable classes (#24529)

Classes that are injectable often have constructors that should not be
called by the application developer. It is the responsibility of the
injector to instantiate the class and the constructor often contains
private implementation details that may need to change.

This commit removes constructors from the docs for API items that are
both:

a) Marked with an injectable decorator (e.g. Injectable, Directive,
Component, Pipe), and
b) Have no constructor description

This second rule allows the developer to override the removal if there
is something useful to say about the constructor.

Note that "normal" classes such as `angimations/HttpHeaders` do not have
their constructor removed, despite (at this time) having no description.

PR Close #24529

petebacondarwin committed:
build(docs-infra): render short description of parameters in API docs (#24537)

PR Close #24537

sjtrimble committed:
feat(docs-infra): Add GitHub and Twitter external icon links to topnav toolbar (#24542)

PR Close #24542

alxhub committed:
fix(ivy): inject() no longer uses default value parameters (#24565)

inject() was changed in da31db7 to not take a default value parameter,
so injectable_compiler_2 should not request the use of one when
using inject().

PR Close #24565

alxhub committed:
fix(ivy): support inputs/outputs in decorator metadata in JIT (#24565)

PR Close #24565

gparlakov committed:
docs(core): typo in static injector tests (#24548)

PR Close #24548

ocombe committed:
feat(ivy): runtime i18n (#24037)

PR Close #24037

benlesh committed:
feat(ivy): Add InheritanceDefinitionFeature to support directive inheritance (#24570)

- Adds InheritanceDefinitionFeature to ivy
- Ensures that lifecycle hooks are inherited from super classes whether they are defined as directives or not
- Directives cannot inherit from Components
- Components can inherit from Directives or Components
- Ensures that Inputs, Outputs, and Host Bindings are inherited
- Ensures that super class Features are run

PR Close #24570

marclaval committed:
fix(ivy): nested ngFor should be supported (#24564)

PR Close #24564

jbogarthyde committed:
docs: fix misdirected group links (#24569)

PR Close #24569

alxhub committed:
feat(ivy): generate ngInjectorDef for @NgModule in AOT mode (#24632)

This change generates ngInjectorDef as well as ngModuleDef for @NgModule
annotated types, reflecting the dual nature of @NgModules as both compilation
scopes and as DI configuration containers.

This required implementing ngInjectorDef compilation in @angular/compiler as
well as allowing for multiple generated definitions for a single decorator in
the core of ngtsc.

PR Close #24632

rkirov committed:
build(bazel): Turning on strictPropertyInitialization for Angular. (#24572)

All errors for existing fields have been detected and suppressed with a
`!` assertion.

Issue/24571 is tracking proper clean up of those instances.

One-line change required in ivy/compilation.ts, because it appears that
the new syntax causes tsickle emitted node to no longer track their
original sourceFiles.

PR Close #24572

pkozlowski-opensource committed:
test(ivy): ngTemplateOutlet runtime integration test (#24587)

PR Close #24587

RicardoVaranda committed:
docs: update Angular CLI option for sourcemaps (#24437)

PR Close #24437

pkozlowski-opensource committed:
fix(ivy): support queries for views inserted in lifecycle hooks (#24587)

Closes #23707

PR Close #24587

brandonroberts committed:
docs: add new Reactive Forms guide (#24578)

PR Close #24578

alxhub committed:
feat(ivy): generate ngInjectorDef for @NgModule in JIT mode (#24632)

This commit takes advantage of the @angular/compiler work for ngInjectorDef
in AOT mode in order to generate the same definition in JIT mode.

PR Close #24632

alxhub committed:
fix(ivy): use closure-safe field name for JIT of ngInjectableDef (#24632)

PR Close #24632

matsko committed:
feat(ivy): properly apply style="", [style], [style.foo] and [attr.style] bindings (#24602)

PR Close #24602

ellamaolson committed:
docs(aio): add elana olson to contributor.json file (#24579)

Add new contributor, elana olson, to the contributors list.

PR Close #24579

gkalpak committed:
build: upgrade AngularJS typings (#24583)

Previously, we were using [@types/angularjs][1], which is deprecated and
outdated (hasn't been updated for over two years). This PR switches to
[@types/angular][2], which is regularly updated (based on the
definitions on [DefinitelyTyped][3]).

[1]: https://www.npmjs.com/package/@types/angularjs
[2]: https://www.npmjs.com/package/@types/angular
[3]: https://github.com/DefinitelyTyped/DefinitelyTyped

PR Close #24583

gkalpak committed:
test(aio): fix upgrade-phonecat examples e2e tests (#24583)

Closes #19625

PR Close #24583

yjaaidi committed:
docs: add guide-angular.wishtack.io to education resources (#24585)

PR Close #24585

gtzinos committed:
docs: update `lowercase` pipe example in "AngularJS to Angular" guide (#24588)

PR Close #24588

gkalpak committed:
build(docs-infra): redirect removed webpack guide to `v5.angular.io` (#24595)

The outdated webpack guide has been removed in #24478, but people might
still try to access it (via direct links or search-engine results).

Instead of returning 404, we will now redirect `/guide/webpack` to the
archived version of the guide at `v5.angular.io/guide/webpack`.

PR Close #24595

alxhub committed:
feat(ivy): able to compile @angular/core with ngtsc (#24677)

@angular/core is unique in that it defines the Angular decorators
(@Component, @Directive, etc). Ordinarily ngtsc looks for imports
from @angular/core in order to identify these decorators. Clearly
within core itself, this strategy doesn't work.

Instead, a special constant ITS_JUST_ANGULAR is declared within a
known file in @angular/core. If ngtsc sees this constant it knows
core is being compiled and can ignore the imports when evaluating
decorators.

Additionally, when compiling decorators ngtsc will often write an
import to @angular/core for needed symbols. However @angular/core
cannot import itself. This change creates a module within core to
export all the symbols needed to compile it and adds intelligence
within ngtsc to write relative imports to that module, instead of
absolute imports to @angular/core.

PR Close #24677

mhevery committed:
release: cut the v6.0.6 release

alxhub committed:
feat(ivy): strip all Angular decorators in compiled classes (#24677)

Previously ngtsc removed the class-level decorators (@Component,
etc) but left all the ancillary decorators (@Input, @Optional,
etc).

This changes the transform to descend into the members of decorated
classes and remove any Angular decorators, not just the class-level
ones.

PR Close #24677

alxhub committed:
build(ivy): run latest build-optimizer on ngtsc compiled code (#24677)

Previously the repo was depending on an old version of build optimizer.
This change updates to the latest (an RC release in the CLI package).

Additionally, this changes the behavior of ng_rollup_bundle to apply
the optimizer to ngtsc compiled code, and configures it to treat the
@angular/compiler package as side-effect-free.

This results in a substantial size reduction of ngtsc compiled code.

PR Close #24677

alxhub committed:
feat(ivy): dummy handler for @Pipe to cause decorator removal (#24677)

Currently ngtsc does not compile @Pipe. This has a side effect
of not removing the @Pipe decorator.

This adds a dummy DecoratorHandler that compiles @Pipe into an
empty ngPipeDef. Eventually this will be replaced with a full
implementation, but for now this solution allows compield code
to be tree-shaken properly.

PR Close #24677

alxhub committed:
test(ivy): symbol extractor should handle different compile options (#24677)

The js_expected_symbol_test implementation extracts symbols names
from a rollup iife bundle. Previously, it only handled the case
with a simple 'var bundle = ...;' statement.

Sometimes, rollup produces a more complex bundle, where the 'bundle'
variable is not the only top-level variable declared in the same
declaration statement. This commit patches the symbol exctractor
to support this more complex case.

Additionally, when the symbol test fails, it prints a command to
accept the symbol diff. This command needs to include the
--define=compile flag to ensure the diff is applied in the same
compile mode as the test was run.

PR Close #24677

alxhub committed:
fix(compiler): fix a few non-tree-shakeable code patterns (#24677)

This change makes @angular/compiler more tree-shakeable by changing
an enum to a const enum and by getting rid of a top-level map that
the tree-shaker was seeing as a reference which caused r3_identifiers
to be retained.

This drops a few hundred bytes of JS from tree-shaken ngtsc compiled
apps.

PR Close #24677

alxhub committed:
test(ivy): move hello_world and todo fully to ngtsc (#24677)

ngtsc is sufficiently capable now that it can compile hello_world
and todo and achieve equivalent results to ngc in ivy (global) mode.

Bundle sizes:
hello_world - 3.0 kB
todo        - 14.7 kB

PR Close #24677

mhevery committed:
release: cut the v6.1.0-beta.2 release

pkozlowski-opensource committed:
refactor(ivy): rename, limit usage of global vars (#24604)

PR Close #24604

mhevery committed:
ci: scripts to review PRs locally (#24623)

PR Close #24623

vikerman committed:
fix(animations): set animations styles properly on platform-server (#24624)

Animations styles weren't getting properly set on platform-server because of erroneous checks and absence of reflection of style property to attribute on the server.

The fix corrects the check for platform and explicitly reflects the style property to the attribute.

PR Close #24624

marclaval committed:
refactor(ivy): insert embedded views immediately (#24629)

PR Close #24629

petebacondarwin committed:
build(docs-infra): implement the 'package' API template (#24631)

PR Close #24631

alxhub committed:
test(ivy): todo app only includes reflect-metadata in JIT mode (#24677)

Previously the todo app imported reflect-metadata, since it is a dependency
of JIT and the todo app tests run in both JIT and AOT modes. However, the
code doesn't get tree-shaken away in AOT mode.

This change adds a target //packages/core/test/bundling/util:reflect_metadata
which, depending on whether the compile flag is in JIT or AOT mode, either
includes reflect-metadata or is a no-op.

Not including reflect-metadata gets the compressed todo bundle down to 12.5 kB.

PR Close #24677

jbogarthyde committed:
docs: add workspace and related cli terms (#24633)

PR Close #24633

mhevery committed:
feat(core): add support for using async/await with Jasmine (#24637)

Example:
```
it('...', await(async() => {
  doSomething();
  await asyncFn();
  doSomethingAfter();
}));
```

PR Close #24637

mhevery committed:
feat(ivy): Support resource resolution in JIT mode. (#24637)

Used to resolve resource URLs on `@Component` when used with JIT compilation.

```
@Component({
  selector: 'my-comp',
  templateUrl: 'my-comp.html', // This requires asynchronous resolution
})
class MyComponnent{
}

// Calling `renderComponent` will fail because `MyComponent`'s `@Compenent.templateUrl`
// needs to be resolved because `renderComponent` is synchronous process.
// renderComponent(MyComponent);

// Calling `resolveComponentResources` will resolve `@Compenent.templateUrl` into
// `@Compenent.template`, which would allow `renderComponent` to proceed in synchronous manner.
// Use browser's `fetch` function as the default resource resolution strategy.
resolveComponentResources(fetch).then(() => {
  // After resolution all URLs have been converted into strings.
  renderComponent(MyComponent);
});

```

PR Close #24637

IgorMinar committed:
feat: typescript 2.9 support (#24652)

PR Close #24652

splincode committed:
docs(changelog): remove repeating blocks (#24654)
PR Close #24654

ocombe committed:
fix(common): use correct currency format for locale de-AT (#24658)

Fixes #24609
PR Close #24658

ocombe committed:
fix(common): use correct ICU plural for locale mk (#24659)
PR Close #24659

jbogarthyde committed:
docs: add tree-shakable providers (#24481)

PR Close #24481

cexbrayat committed:
docs: more info on currency digits (#24661)

Adds an example of using the `currency` pipe with a currency that has no cents like CLP,
which will format the amount with no digits if `digitsInfo` is not provided:

    <!-- outputs CA$14.00 -->
    {{ 14 | currency:'CAD' }}
    <!-- outputs CLP14 -->
    {{ 14 | currency:'CLP' }}

Amends the docs, adds an example and fix an error with a current example.

PR Close #24661

mhevery committed:
ci: fix broken build

gkalpak committed:
refactor(upgrade): fix examples for `strictPropertyInitialization` and remove internal comments (#18487) (#18487)

PR Close #18487

PR Close #18487

gkalpak committed:
fix(ivy): correctly resolve Array  property access (#24664)

PR Close #24664

cexbrayat committed:
docs: typos in directives docs (#24665)

Fixes some typos introduced by #23902

PR Close #24665

pkozlowski-opensource committed:
fix(ivy): report results to appropriate content queries (#24673)

PR Close #24673

alxhub committed:
fix(ivy): export the true ComponentDef/DirectiveDef types (not internal) (#24703)

This was a bug introduced in a previous commit.

PR Close #24703

alxhub committed:
feat(ivy): support pipe compilation from local metadata (#24703)

This updates the r3_pipe_compiler to not depend on global analysis,
and to produce ngPipeDef instructions in the same way that the other
compilers do. It's a precursor to JIT and AOT implementations of
@Pipe compilation.

PR Close #24703

alxhub committed:
feat(ivy): JIT support for compilation of @Pipes (#24703)

Adds support for compiling @Pipe in JIT mode, along with tests
to verify that certain aspects of compilation are correct.

PR Close #24703

alxhub committed:
feat(ivy): AOT support for compilation of @Pipes (#24703)

This commit adds support to ngtsc for compilation of the @Pipe
annotation, including support for pipes in @NgModule scopes.

PR Close #24703

kara committed:
refactor(ivy): remove dynamicParent from LNode (#24678)

PR Close #24678

alxhub committed:
feat(ivy): support templateUrl for ngtsc (#24704)

This commit adds support for templateUrl in component templates within
ngtsc. The compilation pipeline is split into sync and async versions,
where asynchronous compilation invokes a special preanalyze() phase of
analysis. The preanalyze() phase can optionally return a Promise which
will delay compilation until it resolves.

A ResourceLoader interface is used to resolve templateUrls to template
strings and can return results either synchronously or asynchronously.
During sync compilation it is an error if the ResourceLoader returns a
Promise.

Two ResourceLoader implementations are provided. One uses 'fs' to read
resources directly from disk and is chosen if the CompilerHost doesn't
provide a readResource method. The other wraps the readResource method
from CompilerHost if it's provided.

PR Close #24704

jelbourn committed:
fix(platform-browser): add missing deps for HammerGesturesPlugin (#24682)

PR Close #24682

trotyl committed:
fix(common): properly update collection reference in NgForOf (#24684)

closes #24155

PR Close #24684

brandonroberts committed:
docs: include ts-loader as shared example dependency (#24691)

Closes #24671

PR Close #24691

brandonroberts committed:
fix(docs-infra): use clean package.json template when generating zips (#24691)

Closes #24689

PR Close #24691

alan-agius4 committed:
docs: add explanation for `enableResourceInlining` (#24644)

PR Close #24644

brandonroberts committed:
docs(forms): update form builder API reference (#24693)

PR Close #24693

pedroclayman committed:
docs(core): rephrase doc for Injector.get (#24670)

PR Close #24670

kara committed:
fix(ivy): support projecting containers into containers (#24695)

PR Close #24695

jasonaden committed:
release: cut the v6.0.7 release

jasonaden committed:
release: cut the v6.1.0-beta.3 release

robwormald committed:
feat(core): add support for ShadowDOM v1 (#24718)

add a new ViewEncapsulation.ShadowDom option that uses the v1 Shadow DOM API to provide style encapsulation.

PR Close #24718

brandonroberts committed:
docs(forms): update API reference for form validators (#24734)

PR Close #24734

textbook committed:
docs: update HTTP error test example again (#24701)

This has somehow regressed since angular/angular#22844 was merged.

PR Close #24701

kara committed:
refactor(ivy): remove pChild from LNode (#24705)

PR Close #24705

s-gbz committed:
docs: describe rounding behaviour of 'DecimalPipe' (#24303)

PR Close #24303

splincode committed:
docs(changelog): correct inaccuracies (#24713)
PR Close #24713

alxhub committed:
test(common): run common/http tests with Bazel (#24738)

@angular/common/http had tests which were not executed in Bazel. This
commit adds a BUILD.bazel file and ensures the tests pass.

PR Close #24738

alxhub committed:
fix(ivy): generate a type parameter for InjectorDef (#24738)

InjectorDef is parameterized on the type of the injector
configuration class (e.g. the @NgModule decorated type). Previously
this parameter was not included when generating .d.ts files that
contained InjectorDefs.

PR Close #24738

alxhub committed:
feat(ivy): give ngtsc a basic understanding of ModuleWithProviders (#24738)

This commit changes the @NgModule provider to understand that sometimes
an import will resolve to an object instead of a type, and that object
could be of the ModuleWithProviders type. In that case, the 'ngModule'
property is read, and its value used instead.

This still will not handle ModuleWithProviders references across
compilation units; that work is coming in a future PR.

PR Close #24738

alxhub committed:
fix(ivy): *Def types are private (ɵ) symbols (#24738)

On accident a few of the definition types were emitted as public API
symbols. Much of the Ivy API surface is still prefixed with ɵ,
indicating it's a private API. The definition types should be private
for now.

PR Close #24738

alxhub committed:
fix(ivy): remove spurious comma in ngtsc-built .d.ts files (#24738)

On accident a comma was emitted between imports when generating .d.ts
files. This commit removes it.

PR Close #24738

alxhub committed:
build(ivy): enable ngtsc AOT builds for a few packages (#24738)

Turn on AOT builds using ngtsc for:

* animations
* common
* compiler
* compiler-cli
* forms
* platform-browser

PR Close #24738

kara committed:
fix(ivy): correct position for re-projected containers (#24721)

PR Close #24721

petebacondarwin committed:
feat(docs-infra): enable filtering by package on API list page (#24631)

PR Close #24631

alxhub committed:
fix(ivy): strip newlines from selectors in .d.ts files (#24738)

When writing selectors as string literal types in .d.ts files,
strip newlines to avoid generating invalid code. Newlines carry
no meaning in selectors anyway.

PR Close #24738

alxhub committed:
fix(ivy): animations should not be a hard error yet (#24738)

Previously the Ivy template compiler would throw on encountering
an animation binding (e.g. [@anim]). This is unneccessary and
precludes testing existing code. This commit changes the error to a
warning.

PR Close #24738

jbogarthyde committed:
docs: add api doc to commonly queried elements

alxhub committed:
fix(ivy): support zero-argument @NgModule() invocations (#24738)

It's possible to declare an argument-less NgModule:

@NgModule() export class Foo {}

Update the @NgModule compiler to support this usage.

PR Close #24738

gregmagolan committed:
build(bazel): update to latest rules_typescript (#24737)

PR Close #24737

kara committed:
fix(ivy): support projecting into dynamic views (#24752)

PR Close #24752

alxhub committed:
fix(ivy): handle ReadKeyExpr code generation (#24862)

This implements a missing expression type in ngtsc code generation:
that of bracket access to an object property.

PR Close #24862

cexbrayat committed:
test: integration test for TS 2.9.x (#24749)

PR Close #24749

cexbrayat committed:
fix(ivy): pipes are pure by default (#24750)

PR Close #24750

kara committed:
refactor(ivy): replace pNextOrParent with TNode props (#24752)

PR Close #24752

brandonroberts commented:
Hi @loiane, will you rebase on master and remove `(aio)` from the git commit message? Thanks

gregmagolan committed:
build(bazel): update to rule_nodejs 0.10.0 (#24759)

PR Close #24759

gkalpak committed:
build: upgrade jasmine to 2.99.x and fix tests (#19904)

PR Close #19904

trotyl commented on aio/content/marketing/contributors.json:
```diff
@@ -645,5 +645,14 @@
+    "picture": "loiane.jpg",
+    "twitter": "loiane",
+    "website": "https://loiane.com",
+    "bio": "Loiane is a passionate developer and loves writing blog posts and recording free video courses for her training portal https://loiane.training. She is also a GDE (Google Developer Expert) in Angular and Web Technologies.",
```
It may not be a good idea to leave URL in bio, which's neither clickable or selectable, the tile will rotate to photo again once trying to select.

gkalpak committed:
test: make `NgMatchers` type-aware (#19904)

PR Close #19904

gkalpak committed:
refactor: infer type for `it()` assertion functions (#19904)

PR Close #19904

gkalpak committed:
build: upgrade jasmine to 3.1.0 (#19904)

PR Close #19904

gkalpak committed:
build: upgrade karma and related dependencies (#19904)

PR Close #19904

gkalpak committed:
refactor: re-organize and "modernize" cjs-jasmine scripts (#19904)

PR Close #19904

gkalpak committed:
test: run unit tests in random order (#19904)

PR Close #19904

alxhub committed:
fix(ivy): allow building router with ngtsc (#24862)

This commit adds the ivy-local tag to //packages/router. Since the
router depends on //packages/upgrade, it makes that package
compatible with ngtsc as well.

PR Close #24862

gregmagolan committed:
feat(bazel): protractor_web_test_suite for release (#24787)

PR Close #24787

jeffora committed:
fix(compiler-cli): Use typescript to resolve modules for metadata (#22856)

The current module resolution simply attaches .ts to the import/export path, which does
not work if the path is using Node / CommonJS behavior to resolve to an index.ts file.
This patch uses typescript's module resolution logic, and will attempt to load the original
typescript file if this resolution returns a .js or .d.ts file

PR Close #22856

gkalpak committed:
build: make `internal-angular` karma reporter compatible with latest karma (#24803)

Due to changes in karma@1.0.0, `internal-angular` karma reporter stopped
showing browser logs (such as `console.log()` etc.).
Related to d571a5173.

PR Close #24803

gkalpak committed:
build: remove unnecessary `internal-angular` karma reporter (#24803)

The reporter was added in 87d56acda, with the purpose of fixing
source-map paths (which was apparently needed back then). Things have
moved around a lot since then and the custom reporter doesn't seem to be
necessary any more. By removing the reporter, we have one less thing to
worry about while upgrading karma; plus we get improvements in built-in
reporters for free.

Output with the custom reporter:
```
at someMethod (packages/core/.../some-file.ts:13:37)
```

Output with the built-in reporter:
```
at someMethod (packages/core/.../some-file.ts:13.37 <- dist/all/@angular/core/.../some-file.js:1:337)
```

PR Close #24803

alxhub committed:
fix(ivy): unwrap parenthesized or cast expressions for metadata (#24862)

Metadata in Ivy must be literal. For example,

@NgModule({...})

is legal, whereas

const meta = {...};
@NgModule(meta)

is not.

However, some code contains additional superfluous parentheses:

@NgModule(({...}))

It is desirable that ngtsc accept this form of literal object.

PR Close #24862

alxhub committed:
feat(ivy): selector side of ModuleWithProviders via type metadata (#24862)

Within an @NgModule it's common to include in the imports a call to
a ModuleWithProviders function, for example RouterModule.forRoot().
The old ngc compiler was able to handle this pattern because it had
global knowledge of metadata of not only the input compilation unit
but also all dependencies.

The ngtsc compiler for Ivy doesn't have this knowledge, so the
pattern of ModuleWithProviders functions is more difficult. ngtsc
must be able to determine which module is imported via the function
in order to expand the selector scope and properly tree-shake
directives and pipes.

This commit implements a solution to this problem, by adding a type
parameter to ModuleWithProviders through which the actual module
type can be passed between compilation units.

The provider side isn't a problem because the imports are always
copied directly to the ngInjectorDef.

PR Close #24862

alxhub committed:
refactor(ivy): fix an unnecessarily deep import (#24862)

PR Close #24862

alxhub committed:
feat(ivy): statically resolve template expressions (#24862)

This commit adds support for template substitution expressions for
ngtsc static resolution.

PR Close #24862

alxhub committed:
fix(ivy): export InheritDefinitionFeature (#24862)

PR Close #24862

jbogarthyde committed:
docs: correct project definition (#24807)

PR Close #24807

pkozlowski-opensource committed:
refactor(ivy): remove content query creation from directive factories (#24811)

PR Close #24811

gkalpak committed:
docs: fix typo in Universal guide (#24812)

PR Close #24812

vikerman committed:
fix(core): mark NgModule as not the root if APP_ROOT is set to false (#24814)

Tree shakable providers use the APP_ROOT token to determine where to attach themselves. APP_ROOT gets set on NgModule with BrowserModule irrespective of whether it is actually the root(Ex. in case of SSR app where the shell app is first bootstrapped without BrowserModule being the root module).

This change allows a NgModule with BrowserModule to explicitly mark itself as not the root by setting APP_ROOT token to false. This allows tree shakable providers to be attached to the right rott module.

PR Close #24814

vikerman committed:
fix(platform-browser): mark Meta and Title services as tree shakable providers (#24815)

This lets services that use Meta and Title services to be tree shakable and provided in root.

PR Close #24815

alexeagle committed:
build: update to latest nodejs bazel rules (#24817)

PR Close #24817

jasonaden committed:
feat(router): add urlUpdateStrategy allow updating the browser URL at the beginning of navigation (#24820)

Fixes #24616

PR Close #24820

alxhub committed:
fix(ivy): properly inject all special token types (#24862)

Previously ngtsc had a few bugs handling special token types:

* Injector was not properly translated to INJECTOR
* ChangeDetectorRef was not injected via injectChangeDetectorRef()

This commit fixes these two bugs, and also adds a test to ensure
they continue to work correctly.

PR Close #24862

alxhub committed:
fix(ivy): ngInjectorDef should copy full imports/exports nodes (#24862)

@NgModule()s get compiled to two fields: ngModuleDef and ngInjectorDef.
Both fields contain imports, as both selector scopes and injectors have
the concept of composed units of configuration. Previously these fields
were generated by static resolution of imports and exports in metadata.

Support for ModuleWithProviders requires they be generated differently.
ngModuleDef's imports/exports are generated as resolved lists of types,
whereas ngInjectorDef's imports should reflect the raw expressions that
the developer wrote in the metadata.

This change modifies the NgModule handler and properly copies raw nodes
for the imports and exports into the ngInjectorDef.

PR Close #24862

alxhub committed:
fix(ivy): export NgModuleFactory adapter (#24862)

PR Close #24862

alxhub committed:
fix(ivy): ignore imports without ngInjectorDef in r3_injector (#24862)

ngInjectorDef.imports is generated from @NgModule.imports plus
@NgModule.exports. A problem arises as a result, because @NgModule
exports contain not only other modules (which will have ngInjectorDef
fields), but components, directives, and pipes as well. Because of
locality, it's difficult for the compiler to filter these out at
build time.

It's not impossible, but for now filtering them out at runtime will
allow testing of the compiler against complex applications.

PR Close #24862

jbogarthyde committed:
docs: fix bad link (#24825)

PR Close #24825

rkirov committed:
build(bazel): Undo temporary dependency on unleased TS bazel rules (#24826)

Point to a proper new release version 0.15.1.

PR Close #24826

vicb committed:
fix(common): do not round factional seconds (#24831)

fixes #24384

PR Close #24831

vikerman committed:
fix(platform-browser): workaround wrong import path generated by ngc for DOCUMENT (#24830)

palerdot committed:
docs: fix typo in component architecture guide (#24832)

Change the sentence from 'this tells Angular how provide ...' to 'this tells Angular how to provide ...'. The current sentence does not make grammatical sense.

PR Close #24832

MrJean committed:
docs: fix `elements` example (#24840)

PR Close #24840

alexeagle committed:
build: update to Bazel 0.15 (#24841)

PR Close #24841

alexeagle committed:
build: rename angular_devkit dependency to angular_cli (#24842)

PR Close #24842

vicb committed:
fix(common): format fractional seconds (#24844)

fix a bug introduced in #24831

PR Close #24844

matsko committed:
feat(ivy): properly apply class="", [class], [class.foo] and [attr.class] bindings (#24822)

PR Close #24822

gregmagolan committed:
build(bazel): try removing gazelle (#24787)

PR Close #24787

alxhub committed:
fix(ivy): fix a couple issues with Input/Output compilation (#24862)

PR Close #24862

mhevery committed:
release: cut the v6.0.8 release

mhevery committed:
release: cut the v6.1.0-rc.0 release

mhevery committed:
release: cut the v6.0.9 release

Commit:
docs: update component styles doc regarding relative URL (#24471)

Update the documentation to match the CLI mechanics regarding relative URL in link tags.
docs: update info on stylesheet location for CLI


PR Close #24471

MrJean committed:
docs: add e2e tests for `elements` example (#24840)

PR Close #24840

alxhub committed:
feat(ivy): port the static resolver to use the ReflectionHost (#24862)

Previously, the static resolver did its own interpretation of statements
in the TypeScript AST, which only functioned on TypeScript code. ES5
code in particular would not work with the resolver as it had hard-coded
assumptions about AST structure.

This commit changes the resolver to use a ReflectionHost instead, which
abstracts away understanding of the structural side of the AST. It adds 3
new methods to the ReflectionHost in support of this functionality:

* getDeclarationOfIdentifier
* getExportsOfModule
* isClass

PR Close #24862

alxhub committed:
fix(ivy): detect ngOnChanges as a non-static method (#24862)

Previously ngtsc had a bug where it would only detect the presence of
ngOnChanges as a static method. This commit flips the condition and only
recognizes ngOnChanges as a non-static method.

PR Close #24862

alxhub committed:
fix(ivy): references track the identifier they were discovered under (#24862)

Previously, references had the concept of an identifier, but would not
properly detect whether the identifier should be used or not when
generating an expression. This change fixes that logic.

Additionally, now whenever an identifier resolves to a reference (even
one imported from another module) as part of resolving an expression,
the reference is updated to use that identifier. This ensures that for
a class Foo declared in foo.ts, but referenced in an expression in
bar.ts, the Reference returned includes the identifier from bar.ts,
meaning that writing an expression in bar.ts for the Reference will not
generate an import.

PR Close #24862

alxhub committed:
fix(ivy): wrap non-statement assignment expressions in parentheses (#24862)

Previously, when translating an assignment expression (e.g. x = 3), the
translator would always print the statement as X = Y. However, if the
expression is included in a larger expression (X = (Y = Z)), the
translator would print "X = Y = Z" without regard for the outer
expression context.

Now, the translator understands when it's printing an expression
statement (X = Y;) vs an expression in a larger context (X = (Y = Z);)
and encapsulates the latter in parentheses.

PR Close #24862

alxhub committed:
refactor(ivy): run the compiler compliance tests against ngtsc (#24862)

This commit moves the compiler compliance tests into compiler-cli,
and uses ngtsc to run them instead of the custom compilation
pipeline used before. Testing against ngtsc allows for validation
of the real compiler output.

This commit also fixes a few small issues that prevented the tests
from passing.

PR Close #24862

ocombe committed:
test(ivy): run render3 tests with test.sh (#24866)
PR Close #24866

mlichtblau committed:
docs: update i18n doc regarding aot compilation (#24875)

Add missing lines to code example to allow using ng serve with custom i18n configurations.

PR Close #24875

alexeagle committed:
fix(service-worker): don't include sourceMappingURL in ngsw-worker (#24877)

Fixes #23596

PR Close #24877

alxhub committed:
fix(ivy): correctly export all *Def symbols as private (#24862)

Previously, some of the *Def symbols were not exported or were exported
as public API. This commit ensures every definition type is in the
private export namespace.

PR Close #24862

alxhub committed:
fix(ivy): export injectElementRef (#24862)

PR Close #24862

alxhub committed:
fix(ivy): types in .d.ts files should account for generics (#24862)

Ivy definition types have a generic type which specifies the return
type of the factory function. For example:

static ngDirectiveDef<NgForOf, '[ngFor][ngForOf]'>

However, in this case NgForOf itself has a type parameter <T>. Thus,
writing the above is incorrect.

This commit modifies ngtsc to understand the genericness of NgForOf and
to write the following:

static ngDirectiveDef<NgForOf<any>, '[ngFor][ngForOf]'>

PR Close #24862

Goodwine committed:
fix(compiler): i18n_extractor now outputs the correct source file name (#24885)
for non-inline templates

- Non-inline templates used to ouput the path to the component TS file
instead of the path to the original HTML file.
- Inline templates keep the same behavior.

Fixes #24884

PR Close #24885

gkalpak committed:
docs: clean up `elements` example (indentation, import order, etc) (#24840)

PR Close #24840

gkalpak committed:
docs: refactor `elements` example (#24840)

This makes the closing behavior more deterministic, which makes it
easier to be e2e-tested.

PR Close #24840

pkozlowski-opensource committed:
fix(ivy): invoke lifecycle hooks of directives placed on ng-template (#24899)

PR Close #24899

mlichtblau committed:
docs: update i18n with requested changes (#24875)

use more general project name in code example

PR Close #24875

matsko committed:
build: disable IE web worker tests (#24908)

Travis (saucelabs) has been super flaky when running IE
web worker tests lately. This patch temporarily disables
these tests on IE (not edge) until things get more stable.

PR Close #24908

gkalpak committed:
build(docs-infra): add support for examples of type `elements` (#24840)

Examples using `@angular/elements` need to transpile to es2015 for
Custom Elements to work (on browsers that natively support them).

Alternatively, a polyfill would need to be loaded. For now, changing the
transpilation target to es2015 is the simplest solution.

PR Close #24840

gkalpak committed:
test(platform-webworker): avoid flakes due to existing `PlatformRef` (#24916)

PR Close #24916

gkalpak committed:
fix(docs-infra): fix table header layout in API pages (#24919)

PR Close #24919

alexeagle committed:
build: Fix windows tests (#24927)

closes #24927

mprobst committed:
fix(animations): @internal must use JSDoc tags. (#24928)

This change fixes up several comments that accidentally used the JSDoc
tag @internal in regular block comments (`/*` instead of `/**`).

This prevents a problem with Closure Compiler that balks at `@` tags
occuring in regular block comments, because it assumes they were
intended to be tags for the compiler.

When occuring in `/**` JSDoc, tsickle escapes the tags, so they do not
cause problems.

PR Close #24928

Xapuu committed:
docs: remove closing parenthesis from provides guide (#24935)

PR Close #24935

alxhub committed:
fix(ivy): use 'typeof' and 'never' for type metadata (#24862)

Previously ngtsc would use a tuple of class types for listing metadata
in .d.ts files. For example, an @NgModule's declarations might be
represented with the type:

[NgIf, NgForOf, NgClass]

If the module had no declarations, an empty tuple [] would be produced.

This has two problems.

1. If the class type has generic type parameters, TypeScript will
complain that they're not provided.

2. The empty tuple type is not actually legal.

This commit addresses both problems.

1. Class types are now represented using the `typeof` operator, so the
above declarations would be represented as:

[typeof NgIf, typeof NgForOf, typeof NgClass].

Since typeof operates on a value, it doesn't require generic type
arguments.

2. Instead of an empty tuple, `never` is used to indicate no metadata.

PR Close #24862

alxhub committed:
feat(ivy): compile queries in ngtsc (#24862)

This commit adds support for @ContentChild[ren] and @ViewChild[ren] in
ngtsc. Previously queries were ignored.

PR Close #24862

alxhub committed:
feat(ivy): support host bindings in ngtsc (#24862)

This change adds support for host bindings to ngtsc, and parses them
both from decorators and from the metadata in the top-level annotation.

PR Close #24862

jbogarthyde committed:
docs: technical review incorporated (#24744)

closes #24744

matsko committed:
release: cut the v6.1.0-rc.3 release

Note that RC1 and RC2 glitched out midway during the npm release.
Therefore there is only one commit

vicb committed:
ci: remove Tina from pullaprove (#25006)

She has been removed as a collaborator to the project and pullaprove rejects
this config file which still lists her name.

PR Close #25006

vikerman committed:
fix(core): stop reusing provider definitions across NgModuleRef instances (#25022)

Fixes #25018.

Instantiating a NgModuleRef from NgModuleFactory reuses the NgModuleDefinition if it is already present. However the NgModuleDefinition has a providers array which modified when tree shakable providers are instantiated. This corrupts the provider definitions the next time the same factory is used to create a new NgModuleRef - Two provider definitions can end up with the same index anf the injector could potentially return a completely wrong object for a provider token.

This scenario is more likely on the server where the same NgModuleFactory is reused across requests.

The fix clones the cached NgModuleDefinition so that any tree shakable providers added later do not affect the cached copy.

PR Close #25022

brandonroberts commented:
@loiane will you rebase this one again?

loiane commented:
My apologies, just saw the notification. Let me work on it and will let you know once done.

loiane committed:
docs: added Loiane as GDE in contributors

vicb commented:
@loiane seems like something went wrong please rebase or create a new PR. Thanks.

mary-poppins commented:
You can preview bc4f911 at https://pr24423-bc4f911.ngbuilds.io/.
